test/Constraints/diagnostics.swift

```diff
@@ -376,7 +376,7 @@ CurriedClass.method3(c)(32, b: 1)
 _ = CurriedClass.method3(c)
 _ = CurriedClass.method3(c)(1, 2)        // expected-error {{missing argument label 'b:' in call}} {{32-32=b: }}
 _ = CurriedClass.method3(c)(1, b: 2)(32) // expected-error {{cannot call value of non-function type '()'}}
-_ = CurriedClass.method3(1, 2)           // expected-error {{instance member 'method3' cannot be used on type 'CurriedClass'; did you mean to use a value of this type instead?}}
+_ = CurriedClass.method3(1, 2)           // expected-error {{missing argument label 'b:' in call}}
```

Comment:
I'm not actually sure if this is better diagnostic - it's indeed pointing to missing `b:` but the main problem here is that `method3` can't be called on metatype like that...

Reply:
I agree with your, Pavel. The original error message makes more sense. I updated the code to properly handle the case of instance method as curried member on type. Now the PR does not change this file.